c10/util/Unicode.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#if defined(_WIN32)
+#define WIN32_LEAN_AND_MEAN
+#define NOMINMAX
+#define NOGDI
```

Comment:
@peterjc123  I'm a bit confused. Theoretically `WIN32_LEAN_AND_MEAN` should imply `NOGDI`. Why is this still needed?

Reply:
Nope, with `WIN32_LEAN_AND_MEAN`, it doesn't explicitly define `NOGDI`. And `Windows.h` has a reference to "wingdi.h", in `wingdi.h`, if you don't define `NOGDI`, it will add in a lot macros like `ERROR`. That's why the build fails in previous commits.

Reply:
And, that's why we don't like `Windows.h`, which adds in a lot of stuff without user's notice.

Reply:
My next PR will be targeting the `windows.h` problem 😄

Reply:
Here is the whole lists of the `NO` macros for Windows. Maybe this is helpful to you.
https://stackoverflow.com/questions/1394910/how-to-tame-the-windows-headers-useful-defines

Reply:
It seems that your pr about `windows.h` is landed, maybe you should use it here as well.

Reply:
@peterjc123  Thanks. Will do!